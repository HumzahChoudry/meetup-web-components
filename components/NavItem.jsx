import React from 'react';
import Link from 'react-router/lib/Link';
import cx from 'classnames';

/**
 * Navigation item component using SQ2 styles
 * @see {@link http://meetup.github.io/sassquatch2/}
 * @module NavItem
 */
class NavItem extends React.Component {
	render() {
		const {
			className,
			to,
			children,
			...other
		} = this.props;

		const classNames = cx(
			'row-item',
			'row-item--shrink',
			'row-item--alignMiddle',
			'padding--all',
			'text--big',
			className
		);

		return (
			<li
				className={classNames}
				{...other}>
					<Link to={to}>
						{children}
					</Link>
			</li>
		);
	}
}

NavItem.propTypes = {
	to: React.PropTypes.string.isRequired,
};

export default NavItem;
