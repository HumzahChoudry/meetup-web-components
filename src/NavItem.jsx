import React from 'react';
import cx from 'classnames';
import {
	// Chunk,
	// Flex,
	FlexItem,
	// Section,
} from './layoutUtils';
/**
 * Navigation item component using SQ2 styles
 * @see {@link http://meetup.github.io/sassquatch2/}
 * @module NavItem
 */
class NavItem extends React.Component {
	render() {
		const {
			className,
			children,
			...other
		} = this.props;

		const classNames = cx(
			'align--center',
			className
		);

		return (
			<FlexItem
				shrink
				className={classNames}
				{...other}>
					{children}
			</FlexItem>
		);
	}
}

export default NavItem;
