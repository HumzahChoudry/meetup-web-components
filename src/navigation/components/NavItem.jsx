import PropTypes from 'prop-types';
import React from 'react';
import cx from 'classnames';

import Button from '../../forms/Button';
import Dropdown from '../../interactive/Dropdown';
import FlexItem from '../../layout/FlexItem';

const NAV_ITEM_CLASS = 'navItemLink';


export const ActionItem = ({ label, action, labelClassNames }) => (
	<Button reset className={cx(NAV_ITEM_CLASS, 'text--secondary')} onClick={action}>
		<span className={labelClassNames}>{label}</span>
	</Button>
);
export const LinkItem = ({ linkTo, navItemContent, className }) => (
	<a href={linkTo} className={cx(NAV_ITEM_CLASS, className)}>
		{navItemContent}
	</a>
);
export const ContentLoaderItem = ({ navItemContent, onClickAction }) => (
	<Button aria-haspopup reset className={NAV_ITEM_CLASS} onClick={onClickAction}>
		{navItemContent}
	</Button>
);
export const DropdownItem = ({ navItemContent, dropdownContent }) => (
	<Dropdown
		noPortal
		align="right"
		maxWidth="544px"
		minWidth="384px"
		trigger={navItemContent}
		content={dropdownContent}
	/>
);

/**
 * NavItem
 *
 * Renders a dropdown if `dropdownContent` prop is passed,
 * otherwise relies on a `linkTo` prop to render a link item.
 *
 * @param {Object} props - React element props
 * @returns {React.element} - navigation segment of header
 */
export const NavItem = props => {
	const {
		linkTo,
		label,
		shrink,
		className,
		labelClassName,
		linkClassName,
		dropdownContent,
		icon,
		hasUpdates,
		onClickAction,
		onAction,
		...other
	} = props;

	const classNames = {
		navItem: cx('navItem', className),
		label: cx('navItem-label', labelClassName),
	};

	const navItemContent = (
		<div>
			{icon}
			{label && <span className={classNames.label}>{label}</span>}
			{hasUpdates && (
				<span className="counterBadge">
					<span className="visibility--a11yHide">
						Updates
					</span>
				</span>
			)}
		</div>
	);

	const trigger = onClickAction ? (
		<ContentLoaderItem
			onClickAction={onClickAction}
			navItemContent={navItemContent}
		/>
	) : (
			navItemContent
		);

	return (
		<FlexItem shrink={shrink} className={classNames.navItem} {...other}>
			{linkTo && (
				<LinkItem
					className={linkClassName}
					linkTo={linkTo}
					navItemContent={navItemContent}
				/>
			)}
			{dropdownContent && (
				<DropdownItem
					dropdownContent={dropdownContent}
					navItemContent={trigger}
				/>
			)}
			{onAction && (
				<ActionItem
					label={label}
					action={onAction}
					labelClassNames={classNames.label}
				/>
			)}
		</FlexItem>
	);
};

NavItem.propTypes = {
	linkTo: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
	dropdownContent: PropTypes.oneOfType([PropTypes.element, PropTypes.bool]),
	onClickAction: PropTypes.oneOfType([PropTypes.func, PropTypes.bool]),
	label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
	shrink: PropTypes.bool,
	className: PropTypes.string,
	labelClassName: PropTypes.string,
	linkClassName: PropTypes.string,
	icon: PropTypes.element,
	hasUpdates: PropTypes.bool,
};

export default NavItem;
