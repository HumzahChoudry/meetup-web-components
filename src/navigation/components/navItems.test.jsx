import React from 'react';
import { shallow } from 'enzyme';

import { NavItem, LinkItem } from './NavItem';

const MOCK_PROPS = {
	linkTo: '/zombocom/',
	title: 'Just a simple title for a11y',
	label: <p>Anything is possible</p>,
	dropdownContent: <p>I am in a dropdown</p>,
	updatesLabel: 'Updates',
};
const renderComponent = props => shallow(<NavItem {...props} />);

describe('NavItem', () => {
	const navItemBasic = renderComponent({
		linkTo: MOCK_PROPS.linkTo,
		title: MOCK_PROPS.title,
	});
	const navItemWithUpdates = renderComponent({
		linkTo: MOCK_PROPS.linkTo,
		title: MOCK_PROPS.title,
		hasUpdates: true,
		updatesLabel: MOCK_PROPS.updatesLabel,
	});

	const navItemWithLabel = renderComponent({
		linkTo: MOCK_PROPS.linkTo,
		title: MOCK_PROPS.title,
		label: MOCK_PROPS.label,
	});

	const navItemWithDropdown = renderComponent({
		label: MOCK_PROPS.label,
		title: MOCK_PROPS.title,
		dropdownContent: MOCK_PROPS.dropdownContent,
	});

	const navItemWithAction = renderComponent({
		label: MOCK_PROPS.label,
		title: MOCK_PROPS.title,
		onAction: jest.fn(),
	});

	it('does *not* render count bubble, label, or dropdown if no count prop is provided', () => {
		expect(navItemBasic).toMatchSnapshot();
	});

	it('renders count bubble when a count is provided', () => {
		expect(navItemWithUpdates).toMatchSnapshot();
	});

	it('renders a label when provided', () => {
		expect(navItemWithLabel).toMatchSnapshot();
	});

	it('link component renders correct url from prop', () => {
		expect(navItemBasic.find(LinkItem)).toMatchSnapshot();
	});

	it('renders a dropdown when `dropdownContent` is passed as a prop', () => {
		expect(navItemWithDropdown).toMatchSnapshot();
	});

	it('renders an ActionItem when `onAction` is passed as a prop', () => {
		expect(navItemWithAction).toMatchSnapshot();
	});
});
