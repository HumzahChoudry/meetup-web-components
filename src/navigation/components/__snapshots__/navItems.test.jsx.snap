// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem does *not* render count bubble, label, or dropdown if no count prop is provided 1`] = `
<FlexItem
  className="navItem"
>
  <LinkItem
    linkTo="/zombocom/"
    navItemContent={
      <div>
        
      </div>
    }
  />
</FlexItem>
`;

exports[`NavItem link component renders correct url from prop 1`] = `
<LinkItem
  linkTo="/zombocom/"
  navItemContent={
    <div>
      
    </div>
  }
/>
`;

exports[`NavItem renders a dropdown when \`dropdownContent\` is passed as a prop 1`] = `
<FlexItem
  className="navItem"
>
  <DropdownItem
    dropdownContent={
      <p>
        I am in a dropdown
      </p>
    }
    navItemContent={
      <div>
        <span
          className="navItem-label"
          title="Just a simple title for a11y"
        >
          <p>
            Anything is possible
          </p>
        </span>
      </div>
    }
  />
</FlexItem>
`;

exports[`NavItem renders a label when provided 1`] = `
<FlexItem
  className="navItem"
>
  <LinkItem
    linkTo="/zombocom/"
    navItemContent={
      <div>
        <span
          className="navItem-label"
          title="Just a simple title for a11y"
        >
          <p>
            Anything is possible
          </p>
        </span>
      </div>
    }
  />
</FlexItem>
`;

exports[`NavItem renders an ActionItem when \`onAction\` is passed as a prop 1`] = `
<FlexItem
  className="navItem"
>
  <ActionItem
    action={[Function]}
    navItemContent={
      <div>
        <span
          className="navItem-label"
          title="Just a simple title for a11y"
        >
          <p>
            Anything is possible
          </p>
        </span>
      </div>
    }
  />
</FlexItem>
`;

exports[`NavItem renders count bubble when a count is provided 1`] = `
<FlexItem
  className="navItem"
>
  <LinkItem
    linkTo="/zombocom/"
    navItemContent={
      <div>
        <span
          className="counterBadge"
        >
          <span
            className="visibility--a11yHide"
          >
            Updates
          </span>
        </span>
      </div>
    }
  />
</FlexItem>
`;
